internal/controllers/machine/machine_controller_phases.go

```diff
@@ -200,6 +193,12 @@ func (r *Reconciler) reconcileBootstrap(ctx context.Context, cluster *clusterv1.
 	}
 	bootstrapConfig := externalResult.Result
```

Comment:
nit: this is declared here but only used after the added if block below, might be slightly cleaner to inverse them

Reply:
+1 (assuming you mean only moving this single line down)

Although I also like they way we handle the externalResult here entirely after we got it and already map it to a bootstrapConfig.

But no strong opinions

Reply:
Moved the line down to just before it's used.